packages/react/dismissable-layer/src/DismissableLayer.tsx

```diff
@@ -106,14 +113,21 @@ const DismissableLayer = React.forwardRef<DismissableLayerElement, DismissableLa
       if (!event.defaultPrevented) onDismiss?.();
     });
 
-    useBodyPointerEvents({ disabled: disableOutsidePointerEvents });
-
     React.useEffect(() => {
       if (!node) return;
-      if (disableOutsidePointerEvents) context.layersWithOutsidePointerEventsDisabled.add(node);
+      if (disableOutsidePointerEvents) {
+        if (context.layersWithOutsidePointerEventsDisabled.size === 0) {
+          originalBodyPointerEvents = document.body.style.pointerEvents;
+          document.body.style.pointerEvents = 'none';
+        }
+        context.layersWithOutsidePointerEventsDisabled.add(node);
+      }
       context.layers.add(node);
       dispatchUpdate();
-    }, [node, disableOutsidePointerEvents, context]);
+      return () => {
+        resetBodyPointerEvents();
```

Comment:
I'm wondering whether it would be easier to reason by colocating the logic of applying **and** removing in the same place? either inline here, or with two methods where we have declared e.g.  `disableBodyPointerEvents` `restoreBodyPointerEvents`